### PR TITLE
ic-tooltip add disablehoverlistener

### PR DESCRIPTION
**Pull request by GCHQ-Developer-530**
## Summary of the changes
Added disableHover prop that will stop the tooltip appearing on hover, and will require a click to appear. The tooltip will still disappear on mouse leave though.

## Related issue
#243 

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [x] I have manually accessibility tested any changes, if relevant.
- [x] I have ensured any changes match the Figma component library.

### CODE REVIEW

**Review by ad9242 — COMMENTED** [review]
Just one comment really: Does it need to emit an event? Given that it's an @internal event and the listener is the same component instance?
Not against it, but maybe see what others think?

**Comment by GCHQ-Developer-530** [comment]
> Just one comment really: Does it need to emit an event? Given that it's an @internal event and the listener is the same component instance? Not against it, but maybe see what others think?

I'm not sure, I'd based it off other @internal events which still seemed to emit an event, but that might just be my lack of knowledge on this stuff, rather than being best practice.

**Comment by ad9242** [comment]
> > Just one comment really: Does it need to emit an event? Given that it's an @internal event and the listener is the same component instance? Not against it, but maybe see what others think?
> 
> I'm not sure, I'd based it off other @internal events which still seemed to emit an event, but that might just be my lack of knowledge on this stuff, rather than being best practice.

So I think where we emit other @internal events is where they are child components that need to communicate back with the parent. In which case the parent will have a @Listener

**Comment by GCHQ-Developer-530** [comment]
> > > Just one comment really: Does it need to emit an event? Given that it's an @internal event and the listener is the same component instance? Not against it, but maybe see what others think?
> > 
> > 
> > I'm not sure, I'd based it off other @internal events which still seemed to emit an event, but that might just be my lack of knowledge on this stuff, rather than being best practice.
> 
> So I think where we emit other @internal events is where they are child components that need to communicate back with the parent. In which case the parent will have a @Listener

Ah okay, so instead of the listener and emitted event, it could just have this.disableHover && this.showEvents = ["click"];?

File: packages/web-components/src/components/ic-tooltip/ic-tooltip.stories.mdx [hunk 1]
```diff
@@ -31,6 +31,27 @@ import readme from "./readme.md";
...
+      html`<ic-tooltip
+        label="This shows 'Code' has been copied"
+        target="test-button-disable-hover"
+        disable-hover="true"
+        ><button
+          id="test-button-disable-hover"
+          aria-describedby="ic-tooltip-test-button-disable-hover"
+          onclick="navigator.clipboard.writeText('Code')"
+        >
+          Disable Hover
```
Comment (ad9242) [inline packages/web-components/src/components/ic-tooltip/ic-tooltip.stories.mdx:48]: Really minor one, but this implies to me that clicking the button will disable the hover. Maybe "Show tooltip" instead?

**Review by ad9242 — APPROVED** [review]

**Pull request merged**